pkg/extension/sumologicextension/extension.go

```diff
@@ -200,104 +168,146 @@ func (se *SumologicExtension) Start(ctx context.Context, host component.Host) er
+			// might have been removed in Sumo.
+			// Fall back to removing the credentials and recreating them by registering
+			// the collector.
+			if err := se.credentialsStore.Delete(se.hashKey); err != nil {
+				se.logger.Error(
+					"Unable to delete old collector credentials", zap.Error(err),
+				)
+			}
+
+			se.logger.Info("Locally stored credentials invalid. Trying to re-register...",
 				zap.String(collectorNameField, colCreds.Credentials.CollectorName),
+				zap.String(collectorIdField, colCreds.Credentials.CollectorId),
+				zap.Error(errV),
 			)
 		} else {
-			se.logger.Info(
-				"Locally stored credentials found, but clobber flag is set: " +
-					"re-registering the collector",
-			)
-			if colCreds, err = se.registerCollectorWithBackoff(ctx, se.collectorName); err != nil {
-				return credentials.CollectorCredentials{}, false, err
-			}
-			registrationDone = true
-			if err := se.credentialsStore.Store(se.hashKey, colCreds); err != nil {
-				se.logger.Error("Unable to store collector credentials", zap.Error(err))
-			}
-		}
-	} else {
-		se.logger.Info("Locally stored credentials not found, registering the collector")
-		if colCreds, err = se.registerCollectorWithBackoff(ctx, se.collectorName); err != nil {
-			return credentials.CollectorCredentials{}, false, err
-		}
-		registrationDone = true
-		if err := se.credentialsStore.Store(se.hashKey, colCreds); err != nil {
-			se.logger.Error("Unable to store collector credentials", zap.Error(err))
+			se.logger.Info("Locally stored credentials not found, registering the collector")
 		}
 	}
 
-	return colCreds, registrationDone, err
+	colCreds, err = se.getCredentialsByRegistering(ctx)
+	if err != nil {
+		return credentials.CollectorCredentials{}, err
+	}
+
+	return colCreds, nil
+}
+
+func (se *SumologicExtension) getCredentialsByRegistering(ctx context.Context) (credentials.CollectorCredentials, error) {
+	colCreds, err := se.registerCollectorWithBackoff(ctx, se.collectorName)
+	if err != nil {
+		return credentials.CollectorCredentials{}, err
+	}
+	if err := se.credentialsStore.Store(se.hashKey, colCreds); err != nil {
+		se.logger.Error(
+			"Unable to store collector credentials, they will be used now but won't be re-used on next run",
+			zap.Error(err),
+		)
+	}
+
+	se.collectorName = colCreds.CollectorName
+
+	return colCreds, nil
+}
+
+// getLocalCredentials returns the credentials either retrieved from local credentials
+// storage in cases they were available there or it registers the collector and
+// returns the credentials obtained from the API.
+// It also returns a boolean flag indicating if the successful registration
+// with the API took place and an error.
+func (se *SumologicExtension) getLocalCredentials(ctx context.Context) (credentials.CollectorCredentials, error) {
+	if !se.credentialsStore.Check(se.hashKey) {
+		return credentials.CollectorCredentials{},
+			fmt.Errorf("collector credentials not found (hash key: %s)", se.hashKey)
+	}
```

Comment:
Is there anything this check gives us that the `.Get` below doesn't?

Reply:
An explicit check only. Changed to `.Get`